auth-server/test/resolvers/mutation/addressSignup.spec.ts

```diff
@@ -0,0 +1,119 @@
+	const username = 'testuser';
+	const name = 'test name';
+	const network = 'kusama';
+	const address = 'HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F'; //Alice
+	const signMessage = 'da194645-4daf-43b6-b023-6c6ce99ee709';
+	const signature = '0x048ffa02dd58557ab7f7ffb316ac75fa942d2bdb83f4480a6698a1f39d6fa1184dd85d95480bfab59f516de578b102a2b01b81ca0e69134f90e0cd08ada7ca88';
+
+	it('should be able to request signup challenge message', async () => {
+		const result = await addressSignupStart(undefined, { address });
+		const redisSignMessage = await redisGet(getAddressSignupKey(address));
+
+		expect(result.signMessage).to.be.a('string');
+		expect(result.signMessage).to.equal(redisSignMessage);
+	});
+
+	it('should be able to signup with correct signature', async () => {
+		// mock the addressSignupStart
+		await redisSetex(getAddressSignupKey(address), ADDRESS_LOGIN_TTL, signMessage);
+
+		const result = await addressSignupConfirm(undefined, { address, email, name, network, signature, username }, fakectx);
+
+		expect(result.user.id).to.exist;
+		expect(result.token).to.be.a('string');
+
+		const token: any = jwt.decode(result.token);
+		expect(token.email).to.equals(email);
+		expect(token.username).to.equals(username);
+		expect(token.sub).to.equals(`${result.user.id}`);
+		expect(token['https://hasura.io/jwt/claims']['x-hasura-kusama-default']).to.equals(address);
+		expect(token['https://hasura.io/jwt/claims']['x-hasura-user-id']).to.equals(`${result.user.id}`);
+
+		const dbuser = await User
+			.query()
+			.where({ id: result.user.id })
+			.first();
+
+		const dbAddress = await Address
+			.query()
+			.where({ user_id: result.user.id })
+			.first();
+
+		expect(dbuser).to.exist;
+		expect(dbuser?.username).to.equal(username);
+		expect(dbuser?.web3signup).to.be.true;
+
+		expect(dbAddress).to.exist;
+		expect(dbAddress?.address).to.equal(address);
+		expect(dbAddress?.verified).to.be.true;
+		expect(dbAddress?.default).to.be.true;
+	});
+
+	it('should be able to login subsequently with address', async () => {
+		// mock the addressLoginStart
+		await redisSetex(getAddressLoginKey(address), ADDRESS_LOGIN_TTL, signMessage);
+
+		const result = await addressLogin(undefined, { address, signature }, fakectx);
+
+		expect(result.user.id).to.exist;
+		expect(result.token).to.be.a('string');
+
+		const token: any = jwt.decode(result.token);
+		expect(token.email).to.equals(email);
+		expect(token.username).to.equals(username);
+		expect(token.sub).to.equals(`${result.user.id}`);
+		expect(token['https://hasura.io/jwt/claims']['x-hasura-kusama-default']).to.equals(address);
+		expect(token['https://hasura.io/jwt/claims']['x-hasura-user-id']).to.equals(`${result.user.id}`);
+
+		await User
+			.query()
+			.where({ id: result.user.id })
+			.del();
+
+		await Address
+			.query()
+			.where({ user_id: result.user.id })
+			.del();
+	});
+
+	it('should not be able to signup with invalid signature', async () => {
```

Comment:
This test is about expired token, not invalid signature. We should definitely try an invalid signature (before the valid one).

Reply:
added